### PR TITLE
Changing BannerCache into ImageCache

### PR DESCRIPTION
**Pull request by Jousway**
Basicly removes the BannerCache and replaces it with a system that supports more images

The default supported images are
- Banners
- Backgrounds
- CDTItles
- Jackets
- CDImages
- Discs

---

Adds a new line to Themes called
`ImageCache=`

example usage:
`ImageCache="Banner,Background,CDTitle"`

---

Adds new Lua call to sprite called
`LoadFromCache();`

example usage:
```
Def.Sprite{
    OnCommand=function(self) self:LoadFromCache("sType", sPath);
}
```

### CODE REVIEW

**Comment by Jousway** [comment]
I change RString to std::string inside ImageCache
Had to add some extra code like
```
Line 337: RString otImagePath = sImagePath; // Remove this when Global RString to std::string convert.
Line 374: RString otImagePath = sImagePath; RString otError = sError; // Remove this when Global RString to std::string convert.
```
Because they accept only RString, When everything is converted to std::string we can remove these.

File: src/Song.cpp [hunk 1]
```diff
@@ -1420,10 +1383,25 @@ RString Song::GetCacheFile(RString sType)
...
+		return PreDefs[sType.c_str()];	
+	
+	// Get all image files and put them into a vector.
+	vector<RString> song_dir_listing;
+	FILEMAN->GetDirListing(m_sSongDir + "*", song_dir_listing, false, false);
+	vector<RString> image_list;
+	vector<RString> fill_exts = ActorUtil::GetTypeExtensionList(FT_Bitmap);
+	for( std::string Image : song_dir_listing )
+	{
+		std::string FileExt = GetExtension(Image).MakeLower();
```
Comment (martensm) [inline src/Song.cpp]: `transform(begin(Image), end(Image), begin(Image), ::tolower);`

File: src/PrefsManager.h [hunk 2]
```diff
@@ -57,26 +57,16 @@ enum ShowDancingCharacters
 	NUM_ShowDancingCharacters,
 	ShowDancingCharacters_Invalid
 };
-enum BannerCacheMode
+enum ImageCacheMode
```
Comment (martensm) [inline src/PrefsManager.h]: Using an enum class makes this more type safe but it's not that important.

Comment (Jousway) [reply src/PrefsManager.h]: most of the stuff in prefs use it so I'ma leave it in for now, if we are planning to remove it entirly lets remove it then 👍

**Comment by Nickito12** [comment]
Is this 100% merge-ready?

**Pull request merged**